src/cryptonote_core/tx_pool.cpp

```diff
@@ -730,20 +738,33 @@ namespace cryptonote
       // 0 fee transactions are never relayed
       if(!meta.pruned && meta.fee > 0 && !meta.do_not_relay)
       {
-        if (!meta.dandelionpp_stem && now - meta.last_relayed_time <= get_relay_delay(now, meta.receive_time))
-          return true;
-        if (meta.dandelionpp_stem && meta.last_relayed_time < now) // for dandelion++ stem, this value is the embargo timeout
-          return true;
+        const relay_method tx_relay = meta.get_relay_method();
+        switch (tx_relay)
+        {
+          case relay_method::stem:
+          case relay_method::forward:
+            if (meta.last_relayed_time > now)
+              return true; // continue to next tx
+            break;
```

Comment:
This code block here should block the PR from getting merged. My initial thought is to reset the timer for expired `forward` and `stem` txes transactions a bit further in the future to the standard (longer) re-relay timeout. I.e. when the initial forward/stem timer expires, the `txpool` needs to protect the txes from being relayed again too soon. Putting it here will guard against other changes to the codebase.

After Dandelion++, the `set_relayed` method is called asynchronously following a `get_relayable_transactions` call. The `get_relayable_transactions` method could be called again, and core could think it needs to relay `forward` transactions again 1 second later. This cannot happen with Dandelion++ stem, because the tx gets bumped to `fluff`, and is therefore marked as `fluff` synchronously (only Dandelion++ stem is delayed, as it asynchronously gets queued for inspecting local fluffing status).